Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp

```diff
@@ -104,6 +109,33 @@ static void webkit_user_content_manager_class_init(WebKitUserContentManagerClass
             g_cclosure_marshal_VOID__BOXED,
             G_TYPE_NONE, 1,
             WEBKIT_TYPE_JAVASCRIPT_RESULT);
+
+    /**
+     * WebKitUserContentManager::script-message-with-reply-received:
+     * @manager: the #WebKitUserContentManager
+     * @user_script_message: the #WebKitUserScriptMessage holding the value received from the JavaScript world.
+     *
+     * This signal is emitted when JavaScript in a web view calls
+     * <code>window.webkit.messageHandlers.<name>.postMessage()</code>, after registering
+     * <code><name></code> using
+     * webkit_user_content_manager_register_script_message_handler_with_reply()
+     *
+     * The callback function is expected to send a reply that could be handled
+     * asynchronously by taking a reference of the reply object.
+     * If no reply is supplied by the user, an automatic reply with an
+     * undefined value will be sent.
+     *
+     * Since: 2.40
+     */
+    signals[SCRIPT_MESSAGE_WITH_REPLY_RECEIVED] =
+        g_signal_new(
+            "script-message-with-reply-received",
+            G_TYPE_FROM_CLASS(gObjectClass),
+            static_cast<GSignalFlags>(G_SIGNAL_RUN_LAST | G_SIGNAL_DETAILED),
+            0, g_signal_accumulator_true_handled, nullptr,
+            g_cclosure_marshal_generic,
+            G_TYPE_BOOLEAN, 1,
+            WEBKIT_TYPE_USER_SCRIPT_MESSAGE_REPLY);
```

Comment:
I think the signal should receive both the message as WebKitJavaScriptResult and the reply